src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconService.kt

```diff
@@ -129,18 +129,22 @@ class ActivitiesReconService(
       nomisApiService.getPrisonerDetails(compareResults.all())
         .sortedBy { it.bookingId }
         .forEach {
-          telemetryClient.trackEvent(
-            "activity-$type-reconciliation-report-failed",
-            mutableMapOf(
-              "prison" to prisonId,
-              "type" to compareResults.differenceType(it.bookingId),
-              "bookingId" to it.bookingId.toString(),
-              "offenderNo" to it.offenderNo,
-              "location" to it.location,
-            ).apply {
-              date?.also { this["date"] = "$it" }
-            },
-          )
+          if (it.location == prisonId) {
+            telemetryClient.trackEvent(
+              "activity-$type-reconciliation-report-failed",
+              mutableMapOf(
+                "prison" to prisonId,
+                "type" to compareResults.differenceType(it.bookingId),
+                "bookingId" to it.bookingId.toString(),
+                "offenderNo" to it.offenderNo,
+                "location" to it.location,
+              ).apply {
+                date?.also { this["date"] = "$it" }
+              },
+            )
+          } else {
+            log.info("Ignoring failed $type reconciliation for booking ${it.bookingId} at prison $prisonId as the prisoner is now at location ${it.location}")
+          }
```

Comment:
The main change - don't report on reconciliation failures where the prisoner has been moved out of the prison. These are not a problem as NOMIS doesn't pay people who have been moved.